File: pys/mtn.py
import requests
from bs4 import BeautifulSoup

url = 'http://news.mtn.co.kr/newscenter/news_section.mtn?scd=all'

req = requests.get(url)
bs = BeautifulSoup(req.content, 'lxml')

wrapper = bs.find('div', id='articleList')
items = wrapper.find_all('li')

for item in items[:1]:
    a_tag = item.find('h3').find('a')
    title = a_tag.get_text().strip()
    href = 'http://news.mtn.co.kr' + a_tag['href']
    date = item.find('span', class_='newsDate').get_text().strip()

    req2 = requests.get(href)
    bs2 = BeautifulSoup(req2.content, 'lxml')
    wrapper2 = bs2.find('div', id='newsContent')

    contents = ''.join([it.get_text().strip() for it in wrapper2.find_all('div')[:-4]]).replace('\n', '')

    print(title, href, date, contents)

File: crawlers/mtn.go
package crawlers

import (
	"fmt"
	"net/http"
	"strings"

	"flashnews/models"
	"flashnews/utils"

	"github.com/PuerkitoBio/goquery"
)

const mtnCommonURL = "http://news.mtn.co.kr/newscenter/news_section.mtn?scd=all"
const mtnItemURL = "http://news.mtn.co.kr"

type MTN struct{}

func (c MTN) GetName() string {
	return "mtn"
}

func (c MTN) GetList(number int) ([]models.NewsItem, error) {
	// Number
	var _number int
	if number > 15 || number < 1 {
		_number = 15
	} else {
		_number = number
	}
	result := make([]models.NewsItem, _number)

	// Request
	req, err := http.Get(mtnCommonURL)
	if err != nil {
		return result, err
	}
	defer req.Body.Close()

	if req.StatusCode != 200 {
		return result, fmt.Errorf("[ERROR] Request Status Code : %d, %s", req.StatusCode, req.Status)
	}

	// HTML Load
	html, err := goquery.NewDocumentFromReader(req.Body)
	if err != nil {
		return result, err
	}

	// Parsing
	wrapper := html.Find("div#articleList")
	items := wrapper.Find("li")
	items.Each(func(i int, sel *goquery.Selection) {
		if i >= _number {
			return
		}

		aTag := sel.Find("h3").Find("a")
		href, ok := aTag.Attr("href")
		if !ok {
			result[i] = models.NewsItem{}
			return
		}
		url := mtnItemURL + href

		date := sel.Find("span.newsDate").Text()
		title := aTag.Text()
		if err != nil {
			result[i] = models.NewsItem{}
			return
		}

		result[i] = models.NewsItem{
			Title:    title,
			URL:      url,
			Contents: "",
			Datetime: date,
		}
	})

	return result, nil
}

func (c MTN) GetContents(item *models.NewsItem) error {
	// Request
	req, err := http.Get(item.URL)
	if err != nil {
		return err
	}
	defer req.Body.Close()

	if req.StatusCode != 200 {
		return fmt.Errorf("[ERROR] Request Status Code : %d, %s", req.StatusCode, req.Status)
	}

	// HTML Load
	html, err := goquery.NewDocumentFromReader(req.Body)
	if err != nil {
		return err
	}

	// Parsing
	wrapper := html.Find("div#newsContent")
	contents := wrapper.Text()
	removes := wrapper.Find("div")
	removes.Each(func(idx int, sel *goquery.Selection) {
		contents = strings.Replace(contents, sel.Text(), "", -1)
	})
	contents = utils.TrimAll(contents)

	item.Contents = contents
	return nil
}
